skrub/_repr.py

```diff
@@ -0,0 +1,103 @@
+import itertools
+
+import sklearn
+from sklearn.utils.fixes import parse_version
+
+sklearn_version = parse_version(sklearn.__version__)
+
+# TODO: remove when scikit-learn 1.6 is the minimum supported version
+# TODO: subsequently, we should remove the inheritance from _HTMLDocumentationLinkMixin
+# for each estimator then.
+if sklearn_version > parse_version("1.6"):
+    from sklearn.utils._estimator_html_repr import _HTMLDocumentationLinkMixin
```

Comment:
IIUC we will now have 2 of those in every skrub estimator's parents, one directly and one through the BaseEstimator?